src/main/scripts/run_e2e_test.py

```diff
@@ -109,11 +110,16 @@ def run_shell_command(cmd):
 print("ls:", os.listdir())
 
 # Run e2e tests
-print("Preparing e2e framework")
-os.chdir("e2e")
-run_shell_command("mvn clean install")
+if args.framework:
+    print("Preparing e2e framework")
+    os.chdir("e2e")
+    run_shell_command("mvn clean install")
+    os.chdir("../plugin")
+else:
+    os.chdir("plugin")
+    run_shell_command("mvn dependency:purge-local-repository -DmanualInclude=io.cdap.tests.e2e:cdap-e2e-framework")
```

Comment:
Why do we need a purge? We are always running the tests on fresh runner, so should not need it.

Reply:
We are caching `.m2 repo` in workflow. 
https://github.com/cdapio/cdap-e2e-tests/blob/5095fd778008227704ea392da4aa9fa606f31845/.github/workflows/e2e.yml#L46